File: ThamudicEpiPlatform/typescript/ocr_scan.ts
import {readFile} from 'node:fs/promises';
const image=process.argv[2],engine=process.argv[3]??'auto';
if(!image){console.error('usage: ocr_scan image [engine]');process.exit(2)}
const form=new FormData();form.append('file',new Blob([await readFile(image)]),'image');
const r=await fetch(`http://127.0.0.1:8010/api/ocr/scan?engine=${encodeURIComponent(engine)}`,{method:'POST',body:form});
console.log(await r.text());if(!r.ok)process.exit(1);

File: ThamudicEpiPlatform/rust/ocr_scan.rs
use reqwest::blocking::{Client, multipart};
use std::{env,fs};
fn main(){let a:Vec<String>=env::args().collect();if a.len()<2{eprintln!("usage: ocr_scan image [engine]");std::process::exit(2)}let engine=a.get(2).cloned().unwrap_or_else(||"auto".into());let data=fs::read(&a[1]).unwrap();let part=multipart::Part::bytes(data).file_name(a[1].clone()).mime_str("application/octet-stream").unwrap();let form=multipart::Form::new().part("file",part);let r=Client::new().post(format!("http://127.0.0.1:8010/api/ocr/scan?engine={engine}")).multipart(form).send().unwrap();println!("{}",r.text().unwrap());}
